### PR TITLE
TRAFODION-2668 one mxosrvr cant startup, dcsServer down

### PR DESCRIPTION
**Pull request by mashengchen**

### CODE REVIEW

File: dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java [hunk 1]
```diff
@@ -414,32 +431,40 @@ public Boolean call() throws Exception {
...
-                       if (!retryCounter.shouldRetry()) {
-                          throw new IOException("Node " + nid + " is not Up");
-                       } else {
-                           retryCounter.sleepUntilNextRetry();
-                           retryCounter.useRetry();
-                       }
-                   }
-                   serverHandlers[childInstance-1] = new ServerHandler(childInstance);
-                   completionService.submit(serverHandlers[childInstance-1]);
+                        if (!retryCounter.shouldRetry()) {
```
Comment (selvaganesang) [inline dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java]: I think you need to fix retryCounterFactory to do the looping for retry count or get rid of retryCounter completely and use the restartAttempt variable.  For the latter case, the line no 445 shouldn't be using retryCounter.shouldRetry() method.

How was this change unit tested?

File: dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java [hunk 2]
```diff
@@ -400,7 +415,9 @@ public Boolean call() throws Exception {
             // /bin/dcs-daemon script
             // which DOES NOT set childServers count.
             for (int childInstance = 1; childInstance <= childServers; childInstance++) {
-                serverHandlers[childInstance-1] = new ServerHandler(childInstance);
+                ServerHandler serverHandler = new ServerHandler(childInstance);
+                serverHandler.setRestartAttempts(this.restartAttempts);
```
Comment (kevinxu021) [inline dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java]: suggest to add a new constructor new ServerHandler(this.conf, childInstance), then if you add more KVs in the future, the only changes may from ServerHandler

File: dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java (hunk 1 above)
Comment (kevinxu021) [inline dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java]: Agree with Selva, please keep retryCounterFactory  and get rid of new retry logic.

File: dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java [hunk 3]
```diff
@@ -414,32 +431,40 @@ public Boolean call() throws Exception {
...
-                   serverHandlers[childInstance-1] = new ServerHandler(childInstance);
-                   completionService.submit(serverHandlers[childInstance-1]);
+                        if (!retryCounter.shouldRetry()) {
+                            throw new IOException("Node " + nid + " is not Up");
+                        } else {
+                            retryCounter.sleepUntilNextRetry();
+                            retryCounter.useRetry();
+                        }
+                    }
+                    if (isRunning) {
```
Comment (kevinxu021) [inline dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java]: Merge the logic with line 444

File: dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java [hunk 4]
```diff
@@ -414,32 +431,40 @@ public Boolean call() throws Exception {
...
+                        restartAttempts = this.restartAttempts;
+                        LOG.info("mxosrvr " + nid + "," + pid + " still running");
+                    }
+                    else {
+                        restartAttempts--;
+                        LOG.info("mxosrvr " + nid + "," + pid + " exited, restarting, restart time : " + restartAttempts);
+                    }
+                    if (restartAttempts > 0) {
```
Comment (kevinxu021) [inline dcs/src/main/java/org/trafodion/dcs/server/ServerManager.java]: Merge the logic with line 444

**Pull request merged**